libcontainer/nsenter/nsexec.c

```diff
@@ -242,19 +320,75 @@ void nsexec()
 	// We must fork to actually enter the PID namespace, use CLONE_PARENT
 	// so the child can have the right parent, and we don't need to forward
 	// the child's exit code or resend its death signal.
-	child = clone_parent(&env);
+	int child = clone_parent(&env, cloneflags);
 	if (child < 0) {
 		pr_perror("Unable to fork");
 		exit(1);
 	}
+	// if uid_map and gid_map were specified, writes the data to /proc files
+	if (uidmap_start > 0 && uidmap_len > 0) {
+		char buf[PATH_MAX];
+		if (snprintf(buf, sizeof(buf), "/proc/%d/uid_map", child) < 0) {
+			pr_perror("failed to construct uid_map file for %d",
+				  child);
+			exit(1);
+		}
 
-	len = snprintf(buf, sizeof(buf), "{ \"pid\" : %d }\n", child);
+		int fd = open(buf, O_RDWR);
+		writedata(fd, data, uidmap_start, uidmap_len);
```

Comment:
You can write only once into /proc/pid/uid_map, so you should not use writedata() here